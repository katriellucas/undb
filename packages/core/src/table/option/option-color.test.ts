import { OptionColor } from './option-color.js'

describe('create', () => {
  test('should create default option color', () => {
    const oc = OptionColor.create()
    expect(oc).to.be.instanceOf(OptionColor)
    expect(oc.name).to.be.eq('indigo')
    expect(oc.shade).to.be.eq(5)
  })

  test('should create option color with value', () => {
    const oc = OptionColor.create({ name: 'blue', shade: 0 })
    expect(oc).to.be.instanceOf(OptionColor)
    expect(oc.name).to.be.eq('blue')
    expect(oc.shade).to.be.eq(0)
  })

  test('should create option color with only name', () => {
    const oc = OptionColor.create({ name: 'blue' })
    expect(oc).to.be.instanceOf(OptionColor)
    expect(oc.name).to.be.eq('blue')
    expect(oc.shade).to.be.eq(5)
  })

  test('should create option color with only shade', () => {
    const oc = OptionColor.create({ shade: 0 })
    expect(oc).to.be.instanceOf(OptionColor)
    expect(oc.name).to.be.eq('indigo')
    expect(oc.shade).to.be.eq(0)
  })
})

describe('next', () => {
  test('should get next color', () => {
    const color = OptionColor.create()
    expect(color.next()).toMatchInlineSnapshot(`
      {
        "name": "violet",
        "shade": 5,
      }
    `)
  })

  test('should get next color for last color name', () => {
    const color = OptionColor.create({ name: 'orange' })
    expect(color.next().name).toMatchInlineSnapshot('"amber"')
  })
})

describe('createColors', () => {
  test('should create colors if no input', () => {
    const colors = OptionColor.createColors([undefined, undefined])
    expect(colors).to.toMatchSnapshot()
  })

  test('should create colors if has input', () => {
    const colors = OptionColor.createColors([{ name: 'blue' }, { name: 'red' }])
    expect(colors).to.toMatchSnapshot()
  })
})
