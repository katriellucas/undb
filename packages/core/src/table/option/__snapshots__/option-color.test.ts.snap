// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createColors > should create colors if has input 1`] = `
[
  {
    "name": "blue",
    "shade": 5,
  },
  {
    "name": "red",
    "shade": 5,
  },
]
`;

exports[`createColors > should create colors if no input 1`] = `
[
  {
    "name": "indigo",
    "shade": 5,
  },
  {
    "name": "violet",
    "shade": 5,
  },
]
`;
